pymeasure/display/inputs.py

```diff
@@ -240,17 +240,27 @@ def validate(self, text, pos):
     def fixCase(self, text):
         self.lineEdit().setText(text.toLower())
 
+    def toDouble(self, string):
+        value, success = self.validator.locale().toDouble(string)
+        if not success:
+            raise ValueError('String could not be converted to a double')
```

Comment:
I mainly added this, so that `toDouble` has the same behavior as the conversion with `float` when it fails. `valueFromText` still looks for a `ValueError` for when converting the input.